compiler/src/iree/compiler/Dialect/Stream/IR/StreamOpFolders.cpp

```diff
@@ -1888,10 +1888,30 @@ void AsyncStoreOp::getCanonicalizationPatterns(RewritePatternSet &results,
 // stream.async.dispatch
 //===----------------------------------------------------------------------===//
 
+namespace {
+
+struct DeduplicateAsyncDispatchEntryRefs final
+    : public OpRewritePattern<AsyncDispatchOp> {
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AsyncDispatchOp dispatchOp,
+                                PatternRewriter &rewriter) const override {
+    auto originalAttr = dispatchOp.getEntryPointsAttr();
+    auto newAttr = deduplicateArrayElements(originalAttr);
+    if (newAttr == originalAttr)
+      return failure();
+    rewriter.updateRootInPlace(
+        dispatchOp, [&]() { dispatchOp.setEntryPointsAttr(newAttr); });
+    return success();
+  }
+};
+
+} // namespace
+
 void AsyncDispatchOp::getCanonicalizationPatterns(RewritePatternSet &results,
                                                   MLIRContext *context) {
-  // TODO(benvanik): nothing? maybe tied type/lifetime updates?
+  // TODO(benvanik):maybe tied type/lifetime updates?
```

Comment:
ultra nit: space after colon

```suggestion
  // TODO(benvanik): maybe tied type/lifetime updates?
```

(fine to automerge and fix later/never ;P)

Reply:
good catch!